lib/src/rule_creator_component.dart

```diff
@@ -0,0 +1,170 @@
+import 'dart:async';
+import 'dart:html';
+
+import 'package:angular/angular.dart';
+import 'package:angular_forms/angular_forms.dart';
+import 'package:fixnum/fixnum.dart';
+import 'package:ng_bootstrap/ng_bootstrap.dart';
+import 'package:recase/recase.dart';
+import 'package:timezone/timezone.dart';
+import 'package:timezone/data/latest.dart';
+
+import 'data_model/rule.pb.dart';
+import 'service/credential_service.dart';
+import 'service/rule_service.dart';
+
+enum RepeatingScheduleType {
+  everyYear,
+  everyMonth,
+  everyDay,
+  everyHour,
+  everyMinute
+}
+
+@Component(
+  selector: 'rule-creator',
+  templateUrl: 'rule_creator_component.html',
+  directives: [coreDirectives, formDirectives, BsInput, bsAccordionDirectives],
+  providers: [ClassProvider(RuleService)],
+)
+class RuleCreatorComponent {
+  final RuleService _ruleService;
+  final CredentialService _credentialService;
```

Comment:
why this is not within `providers` above

Reply:
Oops, it should be! I'll fix it